src/makeCalculator.js

```diff
@@ -4,7 +4,31 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  return {
+    result: 0,
+    add: function (num) {
+      this.result += num;
+    },
+    subtract: function (num) {
+      this.result -= num;
+    },
+    multiply: function (num) {
+      this.result *= num;
+    },
+    divide: function (num) {
+      this.result /= num;
+    },
+    reset: function () {
+      this.result = 0;
+
+      return this;
+    },
+    operate: function (callback, num) {
+      callback.call(this, num);
+
+      return this;
```

Comment:
The `operate` method should be defined using the method shorthand syntax and it is implemented correctly to allow method chaining.